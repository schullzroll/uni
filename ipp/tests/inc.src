.IPPcode22
DEFVAR GF@a
MOVE GF@a int@0
LABEL repeat
WRITE GF@a
WRITE string@,
ADD GF@a GF@a int@1
JUMPIFNEQ repeat GF@a int@10
CREATEFRAME
DEFVAR TF@a
MOVE TF@a bool@true
PUSHFRAME
WRITE string@abc
WRITE LF@a
WRITE bool@false
EXIT int@0