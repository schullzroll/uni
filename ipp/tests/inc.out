<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode22">
    <instruction order="1" opcode="DEFVAR">
        <arg1 type="var">GF@a</arg1>
    </instruction>
    <instruction order="2" opcode="MOVE">
        <arg1 type="var">GF@a</arg1>
        <arg2 type="int">0</arg2>
    </instruction>
    <instruction order="3" opcode="LABEL">
        <arg1 type="label">repeat</arg1>
    </instruction>
    <instruction order="4" opcode="WRITE">
        <arg1 type="var">GF@a</arg1>
    </instruction>
    <instruction order="5" opcode="WRITE">
        <arg1 type="string">,</arg1>
    </instruction>
    <instruction order="6" opcode="ADD">
        <arg1 type="var">GF@a</arg1>
        <arg2 type="var">GF@a</arg2>
        <arg3 type="int">1</arg3>
    </instruction>
    <instruction order="7" opcode="JUMPIFNEQ">
        <arg1 type="label">repeat</arg1>
        <arg2 type="var">GF@a</arg2>
        <arg3 type="int">10</arg3>
    </instruction>
    <instruction order="8" opcode="CREATEFRAME">
    </instruction>
    <instruction order="9" opcode="DEFVAR">
        <arg1 type="var">TF@a</arg1>
    </instruction>
    <instruction order="10" opcode="MOVE">
        <arg2 type="bool">true</arg2>
        <arg1 type="var">TF@a</arg1>
    </instruction>
    <instruction order="11" opcode="PUSHFRAME">
    </instruction>
    <instruction order="12" opcode="WRITE">
        <arg1 type="string">abc</arg1>
    </instruction>
    <instruction order="13" opcode="CONCAT">
        <arg1 type="var">LF@a</arg1>
        <arg2 type="string">a</arg2>
        <arg3 type="string">b</arg3>
    </instruction>
    <instruction order="14" opcode="WRITE">
        <arg1 type="nil">nil</arg1>
    </instruction>
    <instruction order="15" opcode="EXIT">
        <arg1 type="int">32</arg1>
    </instruction>
</program>
